### PR TITLE
Fix BeitForMaskedImageModeling

### LINKED ISSUE #13235: BeitForMaskedImageModeling forward not using bool_masked_pos
### Who can help
@NielsRogge

## Information

I am reading the code of BEIT, trying to use the `BeitForMaskedImageModeling`, while I found the `bool_masked_pos` is defined but not used in `BeitForMaskedImageModeling`'s forward.

In my understanding, `bool_masked_pos` is used to mask out the input image tokens, thus should be passed to the `self.beit` forward here.
https://github.com/huggingface/transformers/blob/5c6eca71a983bae2589eed01e5c04fcf88ba5690/src/transformers/models/beit/modeling_beit.py#L723

Furthermore, the documentation of the `BeitForMaskedImageModeling`'s `forward` lack the description of the `bool_masked_pos`.

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

Fixes #13235 

I've also added an integration test for BeitForMaskedImageModeling, to make sure it returns the same logits as the original implementation on the same input image.

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Very nice, thanks for working on this @NielsRogge!
